src/psc_datalogger/gui.py

```diff
@@ -129,12 +129,19 @@ def handle_start_stop(self, checked):
     def handle_instrument_changed(self):
         """Handle when any of the instrument configurations change"""
         for i in self.instruments:
-            self.connection_manager.set_instrument(
-                i.instrument_number,
-                i.isChecked(),
-                i.get_address(),
-                i.get_temperature_checked(),
-            )
+            try:
+                self.connection_manager.set_instrument(
+                    i.instrument_number,
+                    i.isChecked(),
+                    i.get_address(),
+                    i.get_temperature_checked(),
+                )
+            except ValueError:
```

Comment:
I can't see what would raise ValueError in the code?

Reply:
It's a little convoluted - `Worker.set_instrument` will see an empty string for `gpib_address` when the "Enable" tickbox is first ticked for an instrument (and so the address field has never had a chance to be filled). This sets the `address` variable to `-1`, which is then passed into `Worker._init_instrument`, which sees that `-1` and raises the `ValueError`